### PR TITLE
Potion Javadoc improvements

### PR DESCRIPTION
**Pull request by riking**
- Document fromDamage() and fromItemStack().
- Remove those needless & stupid linebreaks
- Add @deprecated Javadoc to Tier (so that an explanation shows up).
- Improve the class description

See also: https://github.com/riking/Bukkit/commits/extendpotion-throw
Those JDoc improvements require a code change, and will cause compile warnings. I need to check whether they're ready for a PR still.

### CODE REVIEW

**Comment by Wolvereness** [comment]
Looks good, except for the 1-liner tier javadoc. Could you make it a proper multiline, with the same style as other class javadocs?

**Comment by riking** [comment]
@Wolvereness I saw other deprecation javadocs with that style (see the deprecated constructors). Should I add a description to it? Seems a bit weird, though, to document something we don't want to use :s 

Or do you just want

``` java
/**
 * @deprecated bla bla bla
 */
```

**Comment by Wolvereness** [comment]
The second.

**Comment by riking** [comment]
Heh, messed up 0a715f4bdfaa171642afe9cc7b16c45fef29662e and put it on the wrong branch.. would've had to rebase anyway

**Comment by gravitylow** [comment]
This is just a nitpick thing, but your change on https://github.com/Bukkit/Bukkit-JavaDoc/pull/8/files#L0R13 is technically wrong from a grammatical standpoint. If you have items in a list like that, they should all be the same sort of item. For example, you can't have "I went to the park, the gym, the pool, and called my mom" because the last item is a verb and is not the same as the other items. 

You're defining the potion as something that HAS "type, level, name", but HAS doesn't apply to "can be extended or can be a "splash" potion."

Instead I would consider revising that line to "Represents a potion in Minecraft, with a type, level, and name. The potion can be a "splash" potion or can be extended to other portion forms", or something to that effect. I would argue that you don't really need to explain what values the potion has in its class header, but that's up for debate.

**Comment by riking** [comment]
Yeah, I wasn't exactly liking the wording there either -- thanks for the suggestion!

File: src/main/java/org/bukkit/potion/Potion.java [hunk 1]
```diff
@@ -10,7 +10,7 @@
 import com.google.common.collect.ImmutableList;
 
 /**
- * Represents a minecraft potion
+ * Represents a potion in Minecraft, with a type, level, name, and can be extended or can be a "splash" potion.
```
Comment (Wolvereness) [inline src/main/java/org/bukkit/potion/Potion.java:13]: All of these properties are implicit based on method names. If we need to describe that stuf,f it goes in that particular method.

File: src/main/java/org/bukkit/potion/Potion.java [hunk 2]
```diff
@@ -330,6 +328,7 @@ public ItemStack toItemStack(int amount) {
         return new ItemStack(Material.POTION, amount, toDamageValue());
     }
 
+    /** @deprecated in favor of integers **/
```
Comment (Wolvereness) [inline src/main/java/org/bukkit/potion/Potion.java:331]: This should not be a single line.

File: src/main/java/org/bukkit/potion/Potion.java [hunk 3]
```diff
@@ -273,8 +272,7 @@ public void setTier(Tier tier) {
     /**
      * Sets the {@link PotionType} of this potion.
      *
-     * @param type
-     *            The new type of this potion
+     * @param type The new type of this potion
```
Comment (Wolvereness) [inline src/main/java/org/bukkit/potion/Potion.java:275]: Not a sentence, so the first letter shouldn't be capitalized (this is recurring in other places on your diff as well).

**Comment by Wolvereness** [comment]
Abandoned.

**Pull request closed without merging**